Lib/ufo2ft/featureWriters/kernFeatureWriter.py

```diff
@@ -142,91 +143,58 @@ def partitionByScript(
         """Split a potentially mixed-script pair into pairs that make sense based
         on the dominant script, and yield each combination with its dominant script."""
 
-        # First, partition the pair by their assigned scripts
-        allFirstScripts: dict[tuple[str, ...], list[str]] = {}
-        allSecondScripts: dict[tuple[str, ...], list[str]] = {}
+        # First, partition the pair by their assigned scripts. Glyphs can have
+        # multiple scripts assigned to them (legitimately, e.g. U+0951
+        # DEVANAGARI STRESS SIGN UDATTA, or for random reasons like having both
+        # `sub h by h.sc` and `sub Etaprosgegrammeni by h.sc;`). We duplicate
+        # the glyph name into each script bucket because each script gets its
+        # own lookup and group membership is exclusive per lookup.
```

Comment:
Wait, does this also apply to Common glyphs? We just toss them into all lookups :thinking: I'm confused.